arrows.lic

```diff
@@ -42,47 +61,70 @@ class Arrows
+      bput("get #{@lumber} lumber from #{@bag}", "You get", "What were")
+      bput("combine my lumber", "You combine", "You must")
+      bput("stow lumber", "You put")
+    end
   end
 
   def check_flights
-    return unless bput('tap my flights', 'You tap', 'What were') == 'What were'
-    fput('order 12')
-    fput('order 12')
-    fput('stow flights')
+    return if exists?('flights')
+    flight = find_piece('arrow flights')
+    2.times do
+      order_item(@hometown_data['stock-room'], flight)
+      bput("stow flights", "You put")
+    end
   end
 
   def make_shafts
-    return unless bput('tap my shafts', 'You tap', 'What were') == 'What were'
-    order_lumber if bput("tap my #{lumber} lumber", 'You tap', 'What were') == 'What were'
-    fput("untie my shaper from my #{@belt}")
-    fput("get my #{lumber} lumber from my #{@bag}")
-    bput('shape lumber into arrow shaft', 'Roundtime')
-    fput("tie my shaper to my #{@belt}")
-    fput("stow my shafts in my #{@bag}")
+    return if exists?('shafts')
+    get_item('shaper')
+    bput("get my #{@lumber} lumber from my #{@bag}", "You get", "What were")
+    bput("shape lumber into arrow shaft", 'Roundtime')
+    stow_item('shaper')
+    bput("stow my shafts in my #{@bag}", "You put")
   end
 
   def make_arrowheads
-    new_type = @type
-    new_type == 'boar tusk' if @type == 'boar-tusk'
-    return unless bput('tap my arrowheads', 'You tap', 'I could not') == 'I could not'
-    case bput("get #{@type} from my #{@bag}", 'You get', 'I could not', 'What were')
+    return if exists?('arrowheads')
+    arrow_head = { "cougar-claw" => "curved claw", "boar-tusk" => "boar tusk", "sabertooth" => "ivory fang", "angiswaerd" => "angiswaerd tooth", "hele'la" => "needlesharp tooth", "basilisk" => "basilisk fang", "elsralael" => "elsralael tooth", "soot-stained" => "soot-streaked fang", "ice-adder" => "adder fang", "jagged-horn" => "jagged horn", "drake-fang" => "drake fang" }
+    new_type = arrow_head[@type]
+    case bput("get #{new_type} from my #{@bag}", 'You get', 'I could not', 'What were')
     when /You get/
-      fput("untie my shaper from my #{@belt}")
-      bput("shape my #{new_type} into arrowheads", 'Roundtime')
-      fput("tie my shaper to my #{@belt}")
-      fput("stow my arrowheads in my #{@bag}")
+      get_item('shaper')
+      bput("shape #{new_type} into arrowhead", 'Roundtime', 'You fumble')
+      stow_item('shaper')
+      stow_item('arrowheads')
     when /I could not/, /What were/
       beep
       echo("***Out of #{@type}!***")
       exit
     end
   end
+
+  def clean_up
+    bput("stow shafts", "you pick up", "What were", "Stow what?")
+  end
+
+  def find_piece(match_string)
+    wait_for_script_to_complete('go2', [@hometown_data['stock-room']])
+    echo(match_string)
+    piece = DRC.bput('order', "^\\d+\\)\\.\\s\\s(?:some|a|an|stack of|5 pieces of|10 pieces of|1 batch of straight)?\\s*#{match_string}")
+    piece =~ /^(\d+)/
+    echo(piece)
```

Comment:
💡 Ideally this (and line 115) should be guarded by something like `... if @debug_mode` or such

Reply:
I thought I took those out.

Reply:
I'll remove those in the next iteration.